jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpDateTime.java

```diff
@@ -0,0 +1,229 @@
+        // Note: Calendar.Month fields are zero based.
+        .with("Jan", Calendar.JANUARY + 1)
+        .with("Feb", Calendar.FEBRUARY + 1)
+        .with("Mar", Calendar.MARCH + 1)
+        .with("Apr", Calendar.APRIL + 1)
+        .with("May", Calendar.MAY + 1)
+        .with("Jun", Calendar.JUNE + 1)
+        .with("Jul", Calendar.JULY + 1)
+        .with("Aug", Calendar.AUGUST + 1)
+        .with("Sep", Calendar.SEPTEMBER + 1)
+        .with("Oct", Calendar.OCTOBER + 1)
+        .with("Nov", Calendar.NOVEMBER + 1)
+        .with("Dec", Calendar.DECEMBER + 1)
+        .build();
+
+    private HttpDateTime()
+    {
+    }
+
+    /**
+     * Similar to {@link #parse(String)} but returns unix epoch
+     *
+     * @param datetime the Date/Time to parse.
+     * @return unix epoch in milliseconds, or -1 if unable to parse the input date/time
+     */
+    public static long parseToEpoch(String datetime)
+    {
+        try
+        {
+            Instant instant = parse(datetime);
+            return instant.toEpochMilli();
+        }
+        catch (IllegalArgumentException e)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("Unable to parse Date/Time: {}", datetime, e);
+            return -1;
+        }
+    }
+
+    /**
+     * <p>Parses a Date/Time value</p>
+     *
+     * <p>Supports the following Date/Time formats found in both
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc9110#name-date-time-formats">RFC 9110 (HTTP Semantics)</a> and
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC 6265 (HTTP State Management Mechanism)</a>
+     * </p>
+     *
+     * <ul>
+     *     <li>{@code Sun, 06 Nov 1994 08:49:37 GMT} - RFC 1123 (preferred)</li>
+     *     <li>{@code Sunday, 06-Nov-94 08:49:37 GMT} - RFC 850 (obsolete)</li>
+     *     <li>{@code Sun Nov  6 08:49:37 1994} - ANSI C's {@code asctime()} format</li>
+     * </ul>
+     *
+     * <p>
+     *  Parsing is done according to the algorithm specified in
+     *  <a href="https://datatracker.ietf.org/doc/html/rfc6265#section-5.1.1">RFC6265: Section 5.1.1: Date</a>
+     * </p>
+     *
+     * @param datetime a Date/Time string in a supported format
+     * @return an {@link Instant} parsed from the given string
+     * @throws IllegalArgumentException if unable to parse date/time
+     */
+    public static Instant parse(String datetime)
+    {
+        Objects.requireNonNull(datetime, "Date/Time string cannot be null");
+
+        int year = -1;
+        int month = -1;
+        int day = -1;
+        int hour = -1;
+        int minute = -1;
+        int second = -1;
+
+        try
+        {
+            int tokenCount = 0;
+            StringTokenizer tokenizer = new StringTokenizer(datetime, "\t" + // %x09
+                " !\"#$%&'()*+,-./" + " + " + // %x20-2F
+                ";<=>?@" + // %x3B-40
+                "[\\]^_`" + // %x5B-60
+                "{|}~" // %x7B-7E
+            );
```

Comment:
This would be betters as:
```suggestion
            StringTokenizer tokenizer = new StringTokenizer(datetime, DELIMITERS);
            );
```

with the constant defined by the actual hex code and the characters described in the comment:
```java
    private static final String DELIMITERS = 
          "\u0009" // Horizontal Tab
        + "\u0020\u0021\u0022\u0023\u0024\u0025\u0026\u0027\u0028\u0029\u002A\u002B\u002C\u002D\u002E\u002F" // Space through '/'
        + "\u003B\u003C\u003D\u003E\u003F\u0040" // ';' through '@'
        + "\u005B\u005C\u005D\u005E\u005F\u0060" // '[' through '`'
        + "\u007B\u007C\u007D\u007E"; // '{' through '~'
    }
```

Strictly speaking, the constant is not needed, but I think it reads better.   Eitherway, even without the constant, I prefer it if the string is defined by the hex codes, so I can verify against the RFC without needing to lookup the values